src/clients/db/Repository.ts

```diff
@@ -52,6 +52,27 @@ export default class Repository<T extends AbstractDBModel> {
     return this.find(savedEntity.id)
   }
 
+  async saveOrUpdate(entity: T): Promise<T> {
+    if (!entity.id) {
+      return this.save(entity)
+    }
+
+    try {
+      const existingEntity = await this.find(entity.id)
+      const { id, rev, ...restOfDoc } = existingEntity
+      const entityToUpdate = {
+        _id: id,
+        _rev: rev,
+        ...restOfDoc,
+        ...entity,
+      }
+      await this.db.put(entityToUpdate)
+      return this.find(entity.id)
+    } catch (error) {
+      return this.save(entity)
```

Comment:
Can you explain why saving after an error?

Reply:
So this function is for save and update functionality. We want to save a NEW entity if it didn't' exist in the database and we want to UPDATE an existing entity if it did. 

So the first thing that it does is it tries to find an existing entity. If it didn't find the entity, then it throws an error, which lets us know that we should save a new entity.